launcher/ui/themes/CatPack.cpp

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  Prism Launcher - Minecraft Launcher
+ *  Copyright (c) 2023 Trial97 <alexandru.tripon97@gmail.com>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * This file incorporates work covered by the following copyright and
+ * permission notice:
+ *
+ *      Copyright 2013-2021 MultiMC Contributors
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License");
+ *      you may not use this file except in compliance with the License.
+ *      You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *      Unless required by applicable law or agreed to in writing, software
+ *      distributed under the License is distributed on an "AS IS" BASIS,
+ *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *      See the License for the specific language governing permissions and
+ *      limitations under the License.
+ */
+
+#include "ui/themes/CatPack.h"
+#include <QDate>
+#include <QDir>
+#include <QFileInfo>
+#include "FileSystem.h"
+#include "Json.h"
+#include "ui/themes/ThemeManager.h"
+
+QString BasicCatPack::path()
+{
+    const auto now = QDate::currentDate();
+    const auto birthday = QDate(now.year(), 11, 30);
+    const auto xmas = QDate(now.year(), 12, 25);
+    const auto halloween = QDate(now.year(), 10, 31);
+
+    QString cat = QString(":/backgrounds/%1").arg(m_id);
+    if (std::abs(now.daysTo(xmas)) <= 4) {
+        cat += "-xmas";
+    } else if (std::abs(now.daysTo(halloween)) <= 4) {
+        cat += "-spooky";
+    } else if (std::abs(now.daysTo(birthday)) <= 12) {
+        cat += "-bday";
+    }
+    return cat;
+}
+
+JsonCatPack::JsonCatPack(QFileInfo& manifestInfo) : BasicCatPack(manifestInfo.dir().dirName())
+{
+    QString path = FS::PathCombine("catpacks", m_id);
+
+    if (!FS::ensureFolderPathExists(path)) {
+        themeWarningLog() << "couldn't create folder for catpack!";
+        return;
```

Comment:
this will create a broken instance of `JsonCatPack`

There is no handler for this. Do we need this check in the first place? I think we can assume that the folder path exists anyway.

Reply:
While you did change the logic here, this could still produce an invalid catpack entry. Maybe pass the exception to the caller so that this is not added to the map of catpacks

Reply:
moved the try and catch outside the constructor, if it fails it's not added